File: back-end/subscriptionsWS/routes/moviesRouter.js
const express = require("express");
const route = express.Router();
const moviesBll = require("../BLL/moviesBll")

route.get("/",async (req,res)=>{
    try {
        const movies = await moviesBll.getALLMovies()
        res.status(200).json(movies)

    } catch (error) {
        res.status(500).json(error)
    }
})

route.get("/:id",async (req,res)=>{
    try {
        const { id } = req.params
        const movie = await moviesBll.getOneMovie(id)
        res.status(200).json(movie)

    } catch (error) {
        res.status(500).json(error)
    }
})
route.post("/",async (req,res)=>{
    try {
        const obj  = req.body
        const result = await moviesBll.createMovie(obj)
        res.status(200).json(result)

    } catch (error) {
        res.status(500).json(error)
    }
})
route.put("/:id",async (req,res)=>{
    try {
        const { id } = req.params
        const obj  = req.body
        const result = await moviesBll.updateMovie(id,obj)
        res.status(200).json(result)

    } catch (error) {
        res.status(500).json(error)
    }
})
route.delete("/:id",async (req,res)=>{
    try {
        const { id } = req.params
        const result = await moviesBll.deleteMovie(id)
        res.status(200).json(result)

    } catch (error) {
        res.status(500).json(error)
    }
})


module.exports = route

File: back-end/subWS/routers/moviesRouter.py
from flask import Blueprint,jsonify,request
from Bll.mongoDbBll import MongoDbBll



movies_bll = MongoDbBll("movies")

movies =Blueprint("movies",__name__)

@movies.route("/",methods=["GET"])
def get_all_movies():
    movies = movies_bll.get_all_data()
    return jsonify(movies)

@movies.route("/<string:id>",methods=["GET"])
def get_movie(id):
    movie = movies_bll.get_one_data(id)
    return jsonify(movie)

@movies.route("/",methods=["POST"])
def add_movie():
    obj =request.json
    status = movies_bll.add_data(obj)
    return jsonify(status)

@movies.route("/<string:id>",methods=["PUT"])
def update_movie(id):
    obj =request.json
    obj.pop('_id', None)
    status = movies_bll.update_data(id,obj)
    return jsonify(status)

@movies.route("/<string:id>",methods=["DELETE"])
def delete_movie(id):
    status = movies_bll.delete_data(id)
    return jsonify(status)
